### PR TITLE
magento/devdocs#5245: Hard tabs. Folder: guides/cloud

### PR DESCRIPTION
**Pull request by atwixfirster**
## Purpose of this pull request

This pull request (PR) is related to magento/devdocs#5245: in the folder: `guides/cloud` (v.2.2 and v.2.3)

@meker12 , could you please review!

## Affected DevDocs pages

- ...

## Links to Magento source code

- ...

Thank you!

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by hguthrie** [comment]
The following screenshot shows the result of removing hard spaces from an ordered list with images in a step:

Screen Shot 2019-09-24 at 14 42 00

As you can see, the alignment is off and the list numbering starts over.

**Comment by atwixfirster** [comment]
> The following screenshot shows the result of removing hard spaces from an ordered list with images in a step:
> 
> Screen Shot 2019-09-24 at 14 42 00
> 
> As you can see, the alignment is off and the list numbering starts over.

@hguthrie , could you please verify if guides/v2.2/cloud/before/before-setup-env-2_clone.md looks ok?

I will implement the same styles to the rest pages.

Thank you!

**Comment by hguthrie** [comment]
> @hguthrie , could you please verify if `guides/v2.2/cloud/before/before-setup-env-2_clone.md` looks ok?
> 
> I will implement the same styles to the rest pages.
> 
> Thank you!


@atwixfirster Yes! The topic you fixed looks like it should; the ordered lists are ordered. That is a great help to us. 😄 You're on the right track. If you run into anything weird in fixing the remaining topics, feel free to ask us about it.

**Comment by atwixfirster** [comment]
@hguthrie, PR is ready for your review.

Many thanks!

**Comment by hguthrie** [comment]
@atwixfirster  Okay. At the moment, your branch fails the markdown linter test.
The Cloud guide has received a lot of updates associated with procedure headings, so I fixed the conflicts associated with that but that only helped a little. You may want to run another test.

**Comment by atwixfirster** [comment]
> @atwixfirster Okay. At the moment, your branch fails the markdown linter test.
> The Cloud guide has received a lot of updates associated with procedure headings, so I fixed the conflicts associated with that but that only helped a little. You may want to run another test.

Could you please attach a report with issues? 

Thanks!

**Comment by hguthrie** [comment]
@atwixfirster  It is best if you run the preview and tests in your local environment. Follow the steps defined in the GitHub issue 5245.

**Comment by atwixfirster** [comment]
> @atwixfirster It is best if you run the preview and tests in your local environment. Follow the steps defined in the GitHub issue 5245.

@hguthrie , could you please run tests again?

Thank you!

**Comment by hguthrie** [comment]
@atwixfirster I am seeing artifacts from previous merges to master. And I am seeing changes that have nothing to do with hard tabs. This could be a result of rebasing, but, whatever the issue, we cannot merge this PR and risk the impact to the Cloud guide.

**Comment by atwixfirster** [comment]
> This could be a result of rebasing, but, whatever the issue, we cannot merge this PR and risk the impact to the Cloud guide.

@hguthrie , I hope you joking... 

You requested to fix content after remove hard tabs. And I did that. Now you tell me what you can not merge this PR.

!!!


Please tell me that last you message was a joke.

Thank you!

**Comment by hguthrie** [comment]
Hey Alex, I am not joking, in fact! There are changes in this PR that are not related to the hard tabs issue. Because of the nature of this infrastructure work, we can only accept changes in this PR related to hard tabs.

I know you have worked hard to revisit the changes in Cloud guide, and the changes to ordered lists appear to be fine. The earlier tests were failing because of spacing inserted into blank lines before and after code blocks. However, I reviewed it again and I am still not able to approve this PR for merge. For example, some YAML samples were reformatted with the incorrect spacing, some unnecessary spaces were added to em dashes in lists. I see changes to lexers for some Code blocks, too. These are changes that are not related to the Hard Tab rule issue.

If you’d like to remove changes not applicable to the related issue, I can re-review for merging. Or, if you’d rather start fresh, we can do that too. Please close this and submit another PR.

**Comment by jeff-matthews** [comment]
Ordinarily, I would agree with @hguthrie that we should limit the scope of PRs.  However, these markdown linting issues are unique. 

We (the Magento Docs team) didn't provide sufficient info in the issue to warn contributors about scope, so I'm going to let that slide for this PR.

@atwixfirster, if you will resolve the merge conflicts, I'll re-review your PR. This should resolve all the remaining MD010 errors that are left, which will allow us to enable the rule on production and start making progress on the next linting rule.

**Comment by atwixfirster** [comment]
> @atwixfirster, if you will resolve the merge conflicts, I'll re-review your PR. 

@jeff-matthews  you are the man with a BIG heart. I put a lot of my time to this one.

I've resolved conflicts. Could you please review?

Thank you!

**Comment by atwixfirster** [comment]
> I also see unexpected commits from PRs that were merged to master yesterday. @atwixfirster, did you rebase your branch at some point?

@jeff-matthews , current branch was created from `master`. So, yesterday I used the latest updates from `master` branch.

Did I do something wrong?

Thank you!

**Comment by jeff-matthews** [comment]
I'm not sure. If you rebased after someone else edited your branch, then maybe.

We periodically sync the integration branch with master, so even if you created a working branch from a more recent version of master, those commits should go away after we sync the integration branch.

File: guides/v2.2/cloud/configure/import-url-rewrites.md [hunk 1]
```diff
@@ -15,41 +15,41 @@ This module is available for Magento version 2.2.x only.
...
+   ```bash
+   composer require magento/module-url-rewrite-import-export
+   ```
 
-1.  Configure the `.magento.env.yaml` file deploy stage with cron consumers to run **urlRewriteImport** only.
+1. Configure the `.magento.env.yaml` file deploy stage with cron consumers to run **urlRewriteImport** only.
 
-    ```yaml
-    stage:
+   ```yaml
```
[the remark this answers is not in the document]
Comment (atwixfirster) [reply guides/v2.2/cloud/configure/import-url-rewrites.md]: I've fixed these code blocks

@meker12 , could you please review now? Thank you!

**Comment by jeff-matthews** [comment]
Ok, those commits are now gone. I think this is a result of some idiosyncratic behavior in GitHub. 

This PR originally used `master` as the base branch, but I changed it yesterday to the integration branch. Sometimes, we have to toggle the base branch back and forth to ensure that only the commits from your branch are showing up in the PR.

File: guides/v2.2/cloud/howtos/environment-tutorial-env-merge.md [hunk 2]
```diff
@@ -27,108 +27,106 @@ To get started:
 
 To merge an environment:
 
-1.	Add a file named `test.txt` to the environment root directory.
+1. Add a file named `test.txt` to the environment root directory.
 
-	You can put whatever contents you want; for example, the number `1`
-7.	Save your changes and exit the text editor.
-8.	Add, commit, and push your change to the environment:
+   You can put whatever contents you want; for example, the number `1`
```
[the remark this answers is not in the document]
Comment (atwixfirster) [reply guides/v2.2/cloud/howtos/environment-tutorial-env-merge.md]: I got this file from `master` and re-fixed HARD TAB issue.

Ready for review.

Thank you!

File: guides/v2.2/cloud/configure/import-url-rewrites.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (hguthrie) [reply guides/v2.2/cloud/configure/import-url-rewrites.md]: These appear fine.

File: guides/v2.2/cloud/project/project-webint-snap.md [hunk 3]
```diff
@@ -9,110 +9,110 @@ You can back up and restore specific environments at any time using a snapshot.
...
+1. Create the snapshot.
 
-    ```
-    magento-cloud snapshot:create
-	Creating a snapshot of production
-	Waiting for the snapshot to complete...
-    ```
+   ```
+   magento-cloud snapshot:create
+   Creating a snapshot of production
```
[the remark this answers is not in the document]
Comment (atwixfirster) [reply guides/v2.2/cloud/project/project-webint-snap.md]: Did the same for this one.

Please review. Thank you!

[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.2/cloud/project/project-webint-snap.md]: I don't see a change

[the remark this answers is not in the document]
Comment (atwixfirster) [reply guides/v2.2/cloud/project/project-webint-snap.md]: Indent should be 3 SPACES regarding to https://github.com/magento/devdocs/issues/5253

[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.2/cloud/project/project-webint-snap.md]: The original code block contained indentation inside of it. 

```
magento-cloud snapshot:create
       Creating a snapshot of production
```

[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.2/cloud/project/project-webint-snap.md]: my mistake. no extra indentation needed

**Comment by jeff-matthews** [comment]
Thanks @atwixfirster!

**Comment by atwixfirster** [comment]
> Thanks @atwixfirster!

Thank you @jeff-matthews , for your support here!

Also thank you goes to @meker12  and @hguthrie !

**Pull request merged**